examples/gno.land/p/demo/ui/ui.gno

```diff
@@ -70,56 +98,79 @@ type Link struct {
 
 // TODO: pager
 
-func (l Link) String() string {
-	// TODO: additional arguments.
-	if l.Path != "" && l.URL != "" {
-		panic("a link should have a path OR a URL.")
+func (l Link) String(dom DOM) string {
+	url := ""
+	switch {
+	case l.Path != "" && l.URL != "":
+		panic("a link should have a path or a URL, not both.")
+	case l.Path != "":
+		url = dom.Prefix + l.Path
+	case l.URL != "":
+		url = l.URL
 	}
 
-	if l.Path != "" {
-		// TODO: prefix
-		l.URL = "prefix" + l.Path
-	}
-	return "[" + l.Text + "](" + l.URL + ")"
+	return "[" + l.Text + "](" + url + ")"
 }
 
-func Paragraph(s string) stringer {
-	return Raw{Content: "\n" + s + "\n"}
-}
+type BulletList []domStringer
 
-func BulletList(entries ...stringer) stringer {
+func (bl BulletList) String(dom DOM) string {
 	output := ""
 
-	for _, entry := range entries {
-		output += "- " + entry.String() + "\n"
+	for _, entry := range bl {
+		output += "- " + entry.String(dom) + "\n"
 	}
 
-	return Raw{Content: output}
+	return output
 }
 
-func Text(s string) stringer {
+func Text(s string) domStringer {
 	return Raw{Content: s}
 }
 
-type stringer interface {
-	String() string
+type domStringer interface {
+	String(dom DOM) string
 }
```

Comment:
I think this should be exported if essentially all exported functions use it.

Reply:
done